### PR TITLE
Improve heartbeat handling from client

### PR DESCRIPTION
**Pull request by cangencer**
Also replace heartbeat and partition threads with scheduling service.

### CODE REVIEW

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

File: Hazelcast.Net/Hazelcast.Client.Connection/ClientConnection.cs [hunk 1]
```diff
@@ -133,10 +132,36 @@ internal int Id
...
+        public bool IsHeartBeating
+        {
+            get { return _isHeartBeating; }
+        }
+
         /// <exception cref="System.IO.IOException"></exception>
         public void Close()
         {
-            Release();
+            if (!_live)
```
Comment (ihsandemir) [inline Hazelcast.Net/Hazelcast.Client.Connection/ClientConnection.cs]: do you actually need a compare-and-set here instead?

Comment (cangencer) [inline Hazelcast.Net/Hazelcast.Client.Connection/ClientConnection.cs]: yes good catch, this is replaced with an AtomicBoolean now.

**Comment by cangencer** [comment]
verify

**Comment by asimarslan** [comment]
👍

**Pull request merged**